### ISSUE TITLE
MMEX version 1.6.3 64-bits

### ISSUE THREAD

**Issue by remy12** [moneymanagerex/moneymanagerex#5910]
MMEX version:
 - [ ] 1.6.0
 - [ ] 1.6.1
 - [ ] 1.6.2
 - [ ] 1.6.3 Beta
 - [x] Other (please specify) 1.6.3-Darwing

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [x] Mac OSX
 - [ ] Linux 

# Description of the bug

Please describe your issue with details.
Downloaded the latest version on my Macbook. impossible to open my last database.
Debug report preview:
The following debug report will be generated
"/var/folders/4p/cxrjk47104z8x3nz8rvvdim00000gn/T/MoneyManagerEx_dbgrpt-1431-20230513T142552.zip"
The report contains the files listed below. If any of these files contain private information, please uncheck them and they will be removed from the report.
If you wish to suppress this debug report completely, please choose the "Cancel" button, but be warned that it may hinder improving the program, so if at all possible please do continue with the report generation.
Thank you and we're sorry for the inconvenience!
V
MoneyManagerEx.xl (process context description)
Add screenshot or other files if needed.
![image](https://github.com/moneymanagerex/moneymanagerex/assets/32988255/2d796e1f-5a06-43b6-b7fb-edf28cab2637)
<img width="263" alt="Screenshot 2023-05-13 at 15 06 45" src="https://github.com/moneymanagerex/moneymanagerex/assets/32988255/c7de0722-a038-40e2-b2fe-53761be7b7c2">

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
every Time I want to open a session

Expected result:
open the last database
Actual result:
impossible to open and it shut down
# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by renato-mmex** [comment]
Which MMEX version did you use before.  (1.4?)

**Comment by remy12** [comment]
I think it was 1.4 but cannot remember.Is there a way to fix this?Best regardsRemy Di MeglioOn 13 May 2023, at 16:34, Renato ***@***.***> wrote:﻿
Which MMEX version did you use before.  (1.4?)

**Comment by renato-mmex** [comment]
Maybe this will help: https://github.com/moneymanagerex/moneymanagerex/issues/2353

**Comment by remy12** [comment]
Is the solution to downgrade and re-update or I will never be able to use my 10 years database in the latest version?
I had the version running which came out end January 2023 but I saw an update and after that everything collapsed.

**Comment by remy12** [comment]
￼
When I want to open my database with even the previous version 1.6.0.
Can you send me a debug/patch link that I can download and the steps to get all this fixed?

Thx a lot.
Good evening

**Comment by renato-mmex** [comment]
The MMEX version that came out at the end of january 2023 is from my point of view not the version 1.4
Can you create a list of the existing .mmb and .bak files and paste it here.
This would allow us to trace the history.
Attention enclosures by mail are not displayed in Github.

**Comment by renato-mmex** [comment]
Unfortunately, this information is not enough to make an analysis.

**Comment by remy12** [comment]
Private MMX01-04-2023.mmb_upgrade_v16_2023-05-12.bak

Is this what you refer to?
Cheers

**Comment by renato-mmex** [comment]
@remy12 
Can you create a **list of the existing .mmb and .bak files** and paste it here.
This would allow us to trace the history.
Attention enclosures by mail are not displayed in Github.

**Comment by remy12** [comment]
Hi Renato,

I tried to reopen some previous save database version but did not manage.
Last update was in March 2023 but I recall I already some problems.
It bugged as of 12-05-2023 when upgrading to version 1.6.3.

Private MMX 2023-03-27.mmb_upgrade_v16_2023-05-13.bak
Private MMX 2023-03-27.mmb_upgrade_v15_2023-05-13.bak
Private MMX19042023.mmb
Private MMX19042023.mmb_upgrade_v16_2023-05-13.bak
Private MMX19042023.mmb_upgrade_v15_2023-05-13.bak
PPrivate MMX01-04-2023.mmb
PPrivate MMX01-04-2023.mmb_upgrade_v16_2023-05-12.bak
PPrivate MMX01-04-2023.mmb_upgrade_v15_2023-05-12.bak
Private MM12-05-2023.mmb_upgrade_v16_2023-05-12.bak
Private MM12-05-2023.mmb_upgrade_v15_2023-05-12.bak
Private MM12-05-2023.mmb_update_2023-05-12.bak
Private MMX19042023.mmb_update_2023-05-12.bak
Private MMX19042023.mmb_update_2023-05-10.bak
Private MMX19042023.mmb_update_2023-05-09.bak
PPrivate MMX01-04-2023.mmb_update_2023-04-19.bak
PPrivate MMX01-04-2023.mmb_update_2023-04-18.bak
Private MMX30032023.mmb_update_2023-04-01.bak
Private MMX30032023.mmb_update_2023-03-31.bak
Private MMX 27-03-2023V2.mmb
Private MMX 27-03-2023V2.mmb_update_2023-03-30.bak
Private MMX 27-03-2023V2.mmb_update_2023-03-29.bak
Private MMX 27032023.mmb
Private MMX 27032023.mmb_update_2023-03-27.bak
Private MMX 2023-03-27.mmb_update_2023-03-27.bak
OptimizedDB MMEX.mmb_update_2022-11-26.bak
OptimizedDB MMEX.mmb_update_2022-11-23.bak

Thx for your support.

Best regards
Remy

**Comment by renato-mmex** [comment]
@remy12 
**I have tried to structure the files.**

Private MMX30032023.mmb_update_2023-03-31.bak
Private MMX30032023.mmb_update_2023-04-01.bak
Private MMX19042023.mmb_update_2023-05-09.bak
Private MMX19042023.mmb_update_2023-05-10.bak
Private MMX19042023.mmb_update_2023-05-12.bak
Private MMX19042023.mmb_upgrade_v15_2023-05-13.bak
Private MMX19042023.mmb_upgrade_v16_2023-05-13.bak
Private MMX19042023.mmb

Private MMX 27032023.mmb_update_2023-03-27.bak
Private MMX 27032023.mmb

Private MMX 27-03-2023V2.mmb_update_2023-03-30.bak
Private MMX 27-03-2023V2.mmb_update_2023-03-29.bak
Private MMX 27-03-2023V2.mmb

Private MMX 2023-03-27.mmb_upgrade_v16_2023-05-13.bak
Private MMX 2023-03-27.mmb_upgrade_v15_2023-05-13.bak
Private MMX 2023-03-27.mmb_update_2023-03-27.bak

Private MM12-05-2023.mmb_upgrade_v16_2023-05-12.bak
Private MM12-05-2023.mmb_upgrade_v15_2023-05-12.bak
Private MM12-05-2023.mmb_update_2023-05-12.bak

PPrivate MMX01-04-2023.mmb_update_2023-04-18.bak
PPrivate MMX01-04-2023.mmb_update_2023-04-19.bak
PPrivate MMX01-04-2023.mmb_upgrade_v15_2023-05-12.bak
PPrivate MMX01-04-2023.mmb_upgrade_v16_2023-05-12.bak
PPrivate MMX01-04-2023.mmb


**I find four databases (.mmb), which of them is the one responsible for the error message. ??**
1) Private MMX19042023.mmb
2) Private MMX 27032023.mmb
3) Private MMX 27-03-2023V2.mmb
4) PPrivate MMX01-04-2023.mmb

In addition, there are no .bak files associated with any database ????
Question: was there any error message before debug report ??

**Comment by renato-mmex** [comment]
> Hi Renato,
> The error message came after last upgrade v1.6.3. After opening the app, the error message appeared which i shared on first email.**_The last .mmhe last .mmb is the one from 12-b is the one from 12-05-2023_** which i save specially before upgrading.
> CheersRemy Di MeglioOn

**Please answer my question:**
I find four databases (.mmb), which of them is the one responsible for the error message. ??

Private MMX19042023.mmb
Private MMX 27032023.mmb
Private MMX 27-03-2023V2.mmb
PPrivate MMX01-04-2023.mmb

**Comment by remy12** [comment]
Sorry but it does not upload anything it just shut down and a message suggested to send the report to Apple.
Best regards

**Comment by remy12** [comment]
The last one save before the 12/5 was the one from 19/4.
I change back to my other Mac which still have the version 1.6.0 and it still work with the 19042023.mmb

So the bug comes from any version after this one.
Cheers

**Comment by n-stein** [comment]
> Also, I noticed that the stock market investment portfolio update did not work anymore.

This has been addressed, see #5894.

> I change back to my other Mac which still have the version 1.6.0 and it still work with the 19042023.mmb

The presence of the `*_upgrade_v16_*.bak` files suggests the upgrade to v16 worked. This means your database should open with 1.6.1 with no issues.

When you open your DB with version 1.6.2 or 1.6.3, the system will attempt to upgrade your database to v17. If this fails it will present a message that says "MMEX database upgrade to version 17 failed!" followed by a few lines of text describing the reason the upgrade failed.

It will look something like this (Windows screenshot, but will look similar on Mac):

![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/9db4cdb7-d9f4-4bb2-ab1e-a4a37f6009a8)

Do you see a message like this if you open your database in 1.6.3?

One other thing you can try is to open your database in the latest Beta build of 1.6.4. The upgrade scripts have been adjusted to resolve some common upgrade problems, so it may fix your issue. Note this will upgrade your database to v18 so you will only be able to use version 1.6.4+ on the upgraded database if it is successful.

macOS betas can be found here: https://link.ipx.co.uk/mmex-mac-betas

**Comment by tactilis** [comment]
@remy12
Please could you post your replies and paste any images directly to the github issue here https://github.com/moneymanagerex/moneymanagerex/issues/5910.

If you reply by email, then any images you paste or files you attach are discarded.

**Comment by remy12** [comment]
Screenshot 2023-05-19 at 16.01.24.png

same problem, same error message box and when clicking ok the shutdown of the program.

**Comment by tactilis** [comment]
@remy12
> Screenshot 2023-05-19 at 16.01.24.png

You have to copy and then paste the image (not paste the file name).

Or you can drag the image file into the reply.

**Comment by remy12** [comment]
![image](https://github.com/moneymanagerex/moneymanagerex/assets/32988255/b42b5934-112b-4fcf-b6fe-db51ea3a1bb7)

**Comment by remy12** [comment]
is there a solution to debug the version 1.6.3 and have access again to my database?

**Comment by renato-mmex** [comment]
It is difficult to help you because you do not provide the information needed for troubleshooting. See the post by n-stein.

I could offer you to check your database, for this you would have to send the database to mmex-reg@bluewin.ch. Of course I can guarantee you the necessary confidentiality of the data.

**Comment by n-stein** [comment]
@remy12 

Try this:

1. Download this debug file and unzip it [MMEX_5910_fix.zip](https://github.com/moneymanagerex/moneymanagerex/files/11566311/MMEX_5910_fix.zip)
2. Open your database in 1.6.1. 
3. Go to Tools -> Database -> Database Debug...
4. Load the MMEX_5910_fix.mmdbg file you unzipped in step 1
5. Close MMEX

Then you should be able to open the same database in 1.6.3 or 1.6.4.

**Comment by remy12** [comment]
It did help open the database but the currency and stocks do not update with actuals rates or values.
Is there another debug required?
Thx

**Comment by n-stein** [comment]
No, that is a separate issue. Yahoo Finance has changed their API, we are working on a fix. See #5890, #5894, and recently #5932.

**Comment by remy12** [comment]
sorry but I am not IT expert. I looked at the #... but cannot see what I can download to fix the issue, or the solution is not there yet?

**Comment by reddyshyam** [comment]
> sorry but I am not IT expert. I looked at the #... but cannot see what I can download to fix the issue, or the solution is not there yet?

Hi @remy12, there was a fix put into up coming 1.6.4 which is still in beta stage but that fix stopped working since a day. So there is no solution as such as of now. Once devs decide on how to handle the situation, this topic or the other topics mentioned will be updated. So please keep checking this space for updates.

**Comment by renato-mmex** [comment]
can be closed
